test/integration/ads_integration.cc

```diff
@@ -37,32 +40,46 @@ void AdsIntegrationTest::TearDown() {
 }
 
 envoy::config::cluster::v3::Cluster AdsIntegrationTest::buildCluster(const std::string& name) {
-  return TestUtility::parseYaml<envoy::config::cluster::v3::Cluster>(fmt::format(R"EOF(
+  API_NO_BOOST(envoy::config::cluster::v3::Cluster) cluster;
+  TestUtility::loadFromYaml(
+      fmt::format(R"EOF(
       name: {}
       connect_timeout: 5s
       type: EDS
-      eds_cluster_config: {{ eds_config: {{ ads: {{}} }} }}
+      eds_cluster_config:
+        eds_config:
+          resource_api_version: {}
+          ads: {{}}
       lb_policy: ROUND_ROBIN
       http2_protocol_options: {{}}
     )EOF",
-                                                                                 name));
+                  name, api_version_ == envoy::config::core::v3::ApiVersion::V2 ? "V2" : "V3"),
+      cluster, api_version_ == envoy::config::core::v3::ApiVersion::V2 ? true : false);
+  return cluster;
 }
 
 envoy::config::cluster::v3::Cluster AdsIntegrationTest::buildRedisCluster(const std::string& name) {
-  return TestUtility::parseYaml<envoy::config::cluster::v3::Cluster>(fmt::format(R"EOF(
+  API_NO_BOOST(envoy::config::cluster::v3::Cluster) cluster;
+  TestUtility::loadFromYaml(
+      fmt::format(R"EOF(
       name: {}
       connect_timeout: 5s
       type: EDS
-      eds_cluster_config: {{ eds_config: {{ ads: {{}} }} }}
+      eds_cluster_config:
+        eds_config:
+          resource_api_version: {}
+          ads: {{}}
       lb_policy: MAGLEV
     )EOF",
-                                                                                 name));
+                  name, api_version_ == envoy::config::core::v3::ApiVersion::V2 ? "V2" : "V3"),
+      cluster, api_version_ == envoy::config::core::v3::ApiVersion::V2 ? true : false);
```

Comment:
Should the last arg here be factored out into a `shouldBoost()` predicate util? It comes up again below.

Reply:
addressed